test/e2e/storage/testsuites/base.go

```diff
@@ -129,25 +130,26 @@ func DefineTestSuite(driver TestDriver, tsInits []func() TestSuite) {
 				ginkgo.BeforeEach(func() {
 					// Skip unsupported tests to avoid unnecessary resource initialization
 					suite.SkipRedundantSuite(driver, p)
-					skipUnsupportedTest(driver, p)
```

Comment:
I think it's better to keep this in the base framework so that every single suite doesn't need to remember to call this.

Reply:
We can keep it in base for sure. But the downside is that our TestSuite framework DefineTests() is a public function. And any other people want to use the existing TestSuite they can directly call DefineTests and miss the skipUnsupportedTest part. So it is kind of hard to design the interface in that way and let the user to remember call it before DefineTests. 

From my understanding, each TestSuite should be independent. And Base is just a place that centralize it for K8s e2e purpose. But a caller might not be necessary calling these testsuites from Base. In that case, they will need to call skipUnsupportedTest manually.

Reply:
Hm, that's a fair point.

I guess we already have a testsuite method SkipRedundantSuite that serves a similar purpose. What if we add the common skip util there?

Reply:
Maybe rename it to something like SkipTests

Reply:
Yeah, that sounds good. I will close this PR and create new ones. As this PR has become too huge and its hard to review and test now. I will do the following in later PRs:
1. Redesign the TestSuite interface and skip function.
2. Cleanup some of the utils function
3. Add raw block tests to snapshottable(non-trivial): I took some look and find the current test is not compatible at all with block. So it may need some huge code change.